build_debian.sh

```diff
@@ -318,8 +320,15 @@ sudo LANG=C chroot $FILESYSTEM_ROOT useradd -G sudo,docker $USERNAME -c "$DEFAUL
 echo "$USERNAME:$PASSWORD" | sudo LANG=C chroot $FILESYSTEM_ROOT chpasswd
 
 ## Create redis group
-sudo LANG=C chroot $FILESYSTEM_ROOT groupadd -f redis
-sudo LANG=C chroot $FILESYSTEM_ROOT usermod -aG redis $USERNAME
+sudo LANG=C chroot $FILESYSTEM_ROOT groupadd -f -g $REDIS_USER_GID $REDIS_GROUP
+sudo LANG=C chroot $FILESYSTEM_ROOT usermod -aG $REDIS_GROUP $USERNAME
+
+# Ensure redis gid is 1001.
+# If another GID 1001 already exists, groupadd -f -g will choose another unique GID for redis group.
+redis_gid=$(sudo LANG=C chroot $FILESYSTEM_ROOT getent group $REDIS_GROUP | awk -F: '{print $3}') || true
+if [ "${redis_gid}" != "${REDIS_USER_GID}" ]; then
+    die "Expect Redis group GID ${REDIS_USER_GID}. Current: ${redis_gid}"
```

Comment:
This means that the redis will not load. Isn't this a possible DOS on the redis service?
e.g - an unprivileged user can (via crontab) take ownership of this gid and prevent the redis from loading? (albeit this requires a restart, but still a possible scenario)

Reply:
removed this code block